### PR TITLE
Feature/uncontrast models

### PR DESCRIPTION
**Pull request by dani-efr**

### CODE REVIEW

File: uncontrast_studies/models/effect.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    def __str__(self):
+        return self.type
+
+
+class UnConEffect(models.Model):
+    experiment = models.ForeignKey(
+        null=False, blank=False, to="uncontrast_studies.UnConExperiment", on_delete=CASCADE, related_name="findings"
+    )
+    outcome = models.ManyToManyField(null=False, blank=False, to=UnConEffectOutcome)
```
Comment (alonisser) [inline uncontrast_studies/models/effect.py:17]: in  many to many fields - there is no meaning to null=False,

File: uncontrast_studies/models/effect.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    significance = models.BooleanField(null=False, blank=False)
```
Comment (alonisser) [inline uncontrast_studies/models/effect.py:18]: usually for booleans I prefer a relavant prefix: is, has, etc for example "is_significant"

**Review by alonisser — APPROVED** [review]

**Review by alonisser — APPROVED** [review]

File: uncontrast_studies/models/consciousness_measure.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+
+class UnConsciousnessMeasure(models.Model):
+    experiment = models.ForeignKey(
+        to="uncontrast_studies.UnConExperiment", on_delete=CASCADE, related_name="unconsciousness_measures"
+    )
+    phase = models.ForeignKey(
+        blank=False,
+        null=False,
+        on_delete=CASCADE,
```
Comment (alonisser) [inline uncontrast_studies/models/consciousness_measure.py:40]: I think all those three are "PROTECT" you don't want to allow removing phase, type, subtype if it's in use (unlike the experiment which makes sense)

File: uncontrast_studies/models/consciousness_measure.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        null=True, blank=True, verbose_name="consciousness measure # of trials for the objective measure"
+    )
+    number_of_participants_in_awareness_test = models.PositiveSmallIntegerField(null=True, blank=True)
+    is_cm_same_participants_as_task = models.BooleanField(
+        null=False,
+        blank=False,
+        verbose_name="is consciousness measure taken from the same participants as the main task",
+    )
+    is_performance_above_chance = models.BooleanField(null=False, blank=False)
+    is_trial_excluded_based_on_measure = models.BooleanField(
```
Comment (alonisser) [inline uncontrast_studies/models/consciousness_measure.py:69]: no default here? (also above) you prefer explicit input?

File: uncontrast_studies/models/experiment.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    """
+    This is a major model here, lots of data "belongs" to the experiment in related models, so the experiment is codified there
+    """
+
+    class Meta:
+        ordering = ["id"]
+
+    study = models.ForeignKey(to="studies.Study", on_delete=CASCADE, related_name="uncon_experiments")
+    type = models.PositiveIntegerField(
+        null=False, blank=False, choices=ExperimentTypeChoices.choices, default=ExperimentTypeChoices.NEUROSCIENTIFIC
```
Comment (alonisser) [inline uncontrast_studies/models/experiment.py:18]: same choices as the contrast experiment?

File: uncontrast_studies/models/finding.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+from django.db import models
+from django.db.models import CASCADE
+from simple_history.models import HistoricalRecords
+
+
+class UnConFinding(models.Model):
+    experiment = models.ForeignKey(
+        null=False, blank=False, to="uncontrast_studies.UnConExperiment", on_delete=CASCADE, related_name="findings"
+    )
+    outcome = models.CharField(null=False, blank=False, max_length=250)
```
Comment (alonisser) [inline uncontrast_studies/models/finding.py:10]: looks like a TextField (with no char limitation)

File: uncontrast_studies/models/finding.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    is_significant = models.BooleanField(null=False, blank=False, verbose_name="is effect significant")
+
+    # reported_statistic = models.CharField(null=True, blank=True, max_length=250)
```
Comment (alonisser) [inline uncontrast_studies/models/finding.py:13]: those aren't in use? if so can you please remove them

File: uncontrast_studies/models/paradigm.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+    def __str__(self):
+        return self.name
+
+
+class UnConParadigm(models.Model):
+    experiment = models.ForeignKey(
+        null=False, blank=False, to="uncontrast_studies.UnConExperiment", on_delete=CASCADE, related_name="paradigms"
+    )
+    main = models.ForeignKey(
```
Comment (alonisser) [inline uncontrast_studies/models/paradigm.py:30]: it's really nullable?

**Review by alonisser — APPROVED** [review]
Approved with some comments, but please answer on the processing domain issue

File: uncontrast_studies/models/consciousness_measure.py (hunk 4 above)
Comment (dani-efr) [reply uncontrast_studies/models/consciousness_measure.py:69]: זאת לא העדפה per se, פשוט לא ידעתי את התשובה. שואלת את הצוות שלהם בסלאק.

File: uncontrast_studies/models/finding.py (hunk 6 above)
Comment (dani-efr) [reply uncontrast_studies/models/finding.py:10]: I thought so too at first but the actual data they sent doesn't look like a free text field, more like a short-answer / drop-down options field. I can double check.

File: uncontrast_studies/models/experiment.py (hunk 5 above)
Comment (dani-efr) [reply uncontrast_studies/models/experiment.py:18]: same choices, but should be a different default choice

File: uncontrast_studies/models/suppression_method.py [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    name = models.CharField(null=False, blank=False, max_length=50)
+
+    def __str__(self):
+        return self.name
+
+
+class UnConSuppressionMethodSubType(models.Model):
+    name = models.CharField(null=False, blank=False, max_length=50)
+    parent = models.ForeignKey(
+        null=False, blank=False, related_name="sub_types", to=UnConSuppressionMethodType, on_delete=CASCADE
```
[the remark this answers is not in the document]
Comment (dani-efr) [reply uncontrast_studies/models/suppression_method.py:15]: but then we have a sub-type floating in space with no main type to be attached to...?

File: uncontrast_studies/models/consciousness_measure.py (hunk 4 above)
Comment (dani-efr) [reply uncontrast_studies/models/consciousness_measure.py:69]: they insist on no default

**Pull request merged**